src/mlpack/methods/ann/layer/atrous_convolution_impl.hpp

```diff
@@ -37,6 +37,65 @@ AtrousConvolution<
   // Nothing to do here.
 }
 
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+AtrousConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::AtrousConvolution(
+  const AtrousConvolution& network) :
+    inSize(network.inSize),
+    outSize(network.outSize),
+    kernelWidth(network.kernelWidth),
+    kernelHeight(network.kernelHeight),
+    strideWidth(network.strideWidth),
+    strideHeight(network.strideHeight),
+    inputWidth(network.inputWidth),
+    inputHeight(network.inputHeight),
+    outputWidth(network.outputWidth),
+    outputHeight(network.outputHeight),
+    dilationWidth(network.dilationWidth),
+    dilationHeight(network.dilationHeight),
+    weight(network.weight),
+    bias(network.bias),
```

Comment:
This looks like it will copy both the `weight` and `bias` matrices, but if the owning `FNN/RNN/GAN/BRNN` classes will be calling `Reset()`, then I think that this copy is unnecessary (and computationally expensive too).

Reply:
I think in case of a trained model the weights will be containing some specific values which we will need to copy in another layer. So when I am copying weight and bias we can do it by directly copying weights. What do you suggest??